pkgs/development/python-modules/open-hypergraphs/default.nix

```diff
@@ -0,0 +1,54 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  hatchling,
+  numpy,
+  scipy,
+  hypothesis,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "open-hypergraphs";
+  version = "0.1.2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "statusfailed";
+    repo = "open-hypergraphs";
+    tag = "pypi-${version}";
+    hash = "sha256-ifcQXZDnOvo2XL7WYVFLv2iHWhImUSp3jqAPPYySNjU=";
+  };
+
+  build-system = [
+    hatchling
+  ];
+
+  dependencies = [
+    numpy
+    scipy
+  ];
+
+  pythonRelaxDeps = [
+    "numpy"
+    "scipy"
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    hypothesis
+  ];
+
+  pythonImportsCheck = [
+    "open_hypergraphs"
+  ];
+
+  meta = {
+    description = "An implementation of open hypergraphs for string diagrams";
+    homepage = "https://github.com/statusfailed/open-hypergraphs";
+    changelog = "https://github.com/statusfailed/open-hypergraphs/blob/${src.rev}/CHANGELOG.md";
```

Comment:
I believe relying on src attributes is discouraged. Could we switch to just version instead?

Reply:
I don't think there's consensus on that.

Reply:
In the absence of clear contribution guidelines on this I prefer to leave as-is to avoid oscillating back and forth, I think